### PR TITLE
Initial commit of the scrubber

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR prepares the interface for a new `scrubber` library which we'd use to remove sensitive information at the application level.

See the [design doc](https://www.notion.so/gitpod/Data-Sanitisation-4747f7941ed54b28b2459bab5220febd) for more information.

This library can be imported using `github.com/gitpod-io/gitpod/components/scrubber` as import path. 

>**Note**: Importing this library from outside this repo will download the entire Gitpod repo, a fairly slow operation.

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

**Review by easyCZ — COMMENTED** [review]
What's the reasoning for a new component, rather than `common-go` which this effectively is - common go code

**Comment by svenefftinge** [comment]
Would make sense to replace [redact.go](https://github.com/gitpod-io/gitpod/blob/main/components/common-go/log/redact.go) with this.

File: components/scrubber/config.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package scrubber
+
+import "regexp"
+
+var (
```
Comment (mads-hartmann) [inline components/scrubber/config.go:9]: Have you thought about how we avoid copy-pasting these around? We'll need these in some form or other in:

1. In this file
2. In a similar file for TypeScript
3. In Prometheus drop_label configs
4. In our fluent bit configuration (e.g. if se use [fluent/fluent-plugin-sanitizer](https://github.com/fluent/fluent-plugin-sanitizer))
5. Amazon Macie configuration

Figuring this out is not a blocker for the PR.

**Review by mads-hartmann — APPROVED** [review]
Thanks Chris!

Do I understand correctly that to use this PR to implement the strategy outlined for logs in the design doc [here](https://www.notion.so/gitpod/Data-Sanitisation-4747f7941ed54b28b2459bab5220febd?pvs=4#5b587c4c50a54e818e5a26f332b36a54) we'd use SanitiseHash/SanitiseRedact e.g. [here](https://github.com/gitpod-io/gitpod/blob/main/components/common-go/log/fields.go#L88)?

With the current implementation we might accidentally sanitise values twice. E.g. if we use the scrubber in the telemetry-exporter for analytics and then at a later point also start sanitising the events as we produce them in the Gitpod source code. It might make sense to check if a value is already sanitised before sanitising it.

**Pull request merged**